### DISCUSSION TITLE
How to handle code overflow (going off the page) in PDF output?

### DISCUSSION THREAD

**Discussion by victorfeagins** [quarto-dev/quarto-cli#3693] [Q&A]
I found the feature for HTML outputs here https://quarto.org/docs/output-formats/html-code.html#code-overflow

When I tested the code overflow in the PDF yaml the code will still go off the page so I was curious if there was another way to handle code overflow with PDF's?

```
format: 
   pdf:
       code-overflow: wrap
```

**Comment by mcanouil** [comment]
If the feature is described for HTML and not for other formats, it means the feature is only available/working for the format in which it is described.

Meanwhile, if you are using `knitr` you can apply the same trick as for `rmarkdown` with few modification if you want it to follow Quarto's syntax/style (it should work as is).
See <https://github.com/yihui/knitr-examples/blob/master/077-wrap-output.Rmd>.

**Reply by dtoher** [reply]
Do you know if there are any plans to implement this as an option for pdf (I can use this as a work around at present as am using knitr, but would prefer a solution that doesn't depend on this)?

**Reply by mcanouil** [reply]
No idea. I am not that close to Quarto team to be aware of the roadmap.

Possibly, it could be implemented via a PR by adding an if/else using  `\lstset{breaklines=true, columns=flexible}` (I am not 100 % sure the results of those parameter is satisfying though) in the LaTeX template:

- https://github.com/quarto-dev/quarto-cli/blob/cc21e3522e9a330d0a4181d30afdf385f6142b74/src/resources/formats/pdf/pandoc/template.tex#L206-L214
- https://github.com/quarto-dev/quarto-cli/blob/cc21e3522e9a330d0a4181d30afdf385f6142b74/src/resources/formats/pdf/pandoc/latex.template#L223-L231

**Reply by cscheid** [reply]
No immediate plans on our side. The main issue is that we don't know of a general solution for automatically wrapping code that looks good. In practice, hard-wrapping lines produces code that is as ugly as overflowing code and the solution is style- and language-dependent. As a result, the only good fix is to wrap the code manually. We wish that weren't the case, but we don't currently consider the engineering effort required to make a partial fix worth it.

**Comment by jkylearmstrong** [comment]
@dtoher Did you find a solution to your issue? I attempted the suggestion of [https://github.com/yihui/knitr-examples/blob/master/077-wrap-output.Rmd](https://github.com/yihui/knitr-examples/blob/master/077-wrap-output.Rmd) however, when I try to render with Quarto I get an error:

```{r}
Error in hook_output(x, options) : could not find function "hook_output"
Calls: .main ... <Anonymous> -> lapply -> FUN -> <Anonymous> -> <Anonymous>
Execution halted
```

**Reply by baptiste** [reply]
have you tried this? https://github.com/quarto-dev/quarto-cli/discussions/4121#discussioncomment-4800510

**Reply by jkylearmstrong** [reply]
Thank you, this seems to work pretty well.

**Comment by mcanouil** [comment]
Closing this thread in favour of #4121.
